src/client/eppo-client.ts

```diff
@@ -651,16 +652,39 @@ export default class EppoClient {
   }
 
   private logBanditAction(banditEvent: IBanditEvent): void {
+    const subjectKey = banditEvent.subject;
+    const flagKey = banditEvent.featureFlag;
+    const banditKey = banditEvent.bandit;
+    const actionKey = banditEvent.action ?? '__eppo_no_action';
+
+    // What our bandit assignment cache cares about for avoiding logging duplicate bandit assignments,
+    // if one is active. Like the flag assignment cache, entries are only stored for a given flag
+    // and subject. However, Bandit and action keys are also used for determining assignment uniqueness.
+    // This means that if a flag's bandit assigns one action, and then later a new action, the first
+    // one will be evicted from the cache. If later assigned again, it will be treated as new.
```

Comment:
nit: this comment seems misplaced. The best place for it is calculation of cache key and value (as these are controlling this behavior). If cache key or value change, this comment could easily become outdated.

Reply:
Yeah I was just trying to explain what `banditAssignmentCacheProperties` is and how it will be used, but agree the comment could get stale. I'll experiment with better spatially colocating it.

Reply:
I would move cache entry construction inside the caching module. Otherwise, it looks like a leaking implementation detail — the caller needs to know what keys are used in caching and copy them into `banditAssignmentCacheProperties`. It would be simpler if the cache was accepting the event

Reply:
I found a good home for this in `abstract-assignment-cache.ts` after all! 🎉